src/instruments/src/EFB/Ground/Ground.tsx

```diff
@@ -17,47 +17,142 @@
  */
 
 import React from 'react';
-import { IconCornerDownLeft, IconCornerDownRight, IconArrowDown, IconHandStop, IconTruck, IconBriefcase, IconBuildingBridge, IconBuildingArch, IconArchive } from '@tabler/icons'
+import { IconCornerDownLeft, IconCornerDownRight, IconArrowDown, IconHandStop, IconTruck, IconBriefcase, IconBuildingArch, IconArchive } from '@tabler/icons'
 
 import './Ground.scss'
 import fuselage from '../Assets/320neo-outline-upright.svg'
+import {setSimVar, getSimVar} from '../../util.mjs'
 
 type GroundProps = {}
 
-type GroundState = {}
+type GroundState = {
+    tugActive: boolean;
+    activeButton: string,
+}
 
 class Ground extends React.Component<GroundProps, GroundState> {
 
+    constructor(props: GroundProps) {
+        super(props);
+
+        this.state = {
+            tugActive: false,
+            activeButton: 'none',
+        };
+      }
+
+    toggleGroundAction(action: GroundServices) {
+            setSimVar(action, "1", "boolean");
+    }
+
+    setTugHeading(action: GroundServices, direction: number) {
+            if (!this.state.tugActive) {
+                this.togglePushback();
+            } else {
+                const tugHeading = this.getTugHeading(direction);
+                // KEY_TUG_HEADING is an unsigned integer, so let's convert
+                setSimVar(action, (tugHeading * 11930465) & 0xffffffff, "UINT32");
+            }
+    }
+
+    getTugHeading(value: number): number {
+        const currentHeading = getSimVar("PLANE HEADING DEGREES TRUE", "degrees");
+        return (currentHeading  + value) % 360;
+    }
+
+    togglePushback() {
```

Comment:
I think there's an issue here where if you hit the stop button twice, this.state.tugActive will be set to true, but the tug won't be moving the aircraft. I recommend accepting a boolean parameter for the desired tug state, and not toggling the state parameter

Reply:
Good point, thanks! Will adapt this :)